feat: add unit test for app store

nikaaru opened the pull request:
# Summary

**vitest** is configured to work with **zustand** And a series of unit tests for the App Store have been added to the widget


# How did you test this change?

You can run the `yarn test` command and all the tests should pass correctly.

To see more details, you can enter the embedded project with the `cd widget/embedded/` command and run the `yarn test ` command again.


# Checklist:

- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] Implemented a user interface (UI) change, referencing our Figma design to ensure pixel-perfect precision.

yeager-eren commented:
I've changed your original code to keep only those who should be kept and added another tool to generate fake data. I didn't know what exactly should be result of some of the tests, i made them `skip`, please make sure you will check them. I also put some `TODO:` and `.todo` please make sure you won't miss them.

Besides of this PR, I suggest to take a look at (your previous PRs)[https://github.com/rango-exchange/rango-client/pull/634] and make sure you've covered all the cases mentions in that PR.

nikaaru commented on package.json:
```diff
@@ -47,8 +47,8 @@
-    "test": "vitest",
-    "test:coverage": "vitest run --coverage",
+    "test": "nx run-many --target=test",
+    "test:coverage": "nx run-many --target=test:coverage",
```
I used the **vitest** command and it worked.

nikaaru commented:
> I've changed your original code to keep only those who should be kept and added another tool to generate fake data. I didn't know what exactly should be result of some of the tests, i made them `skip`, please make sure you will check them. I also put some `TODO:` and `.todo` please make sure you won't miss them.
> 
> Besides of this PR, I suggest to take a look at (your previous PRs)[https://github.com/[/pull/634](https://github.com/rango-exchange/rango-client/pull/634)] and make sure you've covered all the cases mentions in that PR.

I fixed all the todos.
But I didn't understand what is your point about this line:

`    // TODO: write a test util to update config.
`
I think it will be cleaner to update the config in the test file.

yeager-eren commented on widget/embedded/src/test-utils/constants.ts:
```diff
@@ -0,0 +1,132 @@
+export const TOKENS_COUNT = 20;
```
Let's keep related pieces together. Move back this to `fixtures`. Constants doesn't make any sense in context of `test utils` and it's part of `fixture` actually.

nikaaru replied:
fixed

yeager-eren commented:
> > I've changed your original code to keep only those who should be kept and added another tool to generate fake data. I didn't know what exactly should be result of some of the tests, i made them `skip`, please make sure you will check them. I also put some `TODO:` and `.todo` please make sure you won't miss them.
> > Besides of this PR, I suggest to take a look at (your previous PRs)[https://github.com/[/pull/634](https://github.com/rango-exchange/rango-client/pull/634)] and make sure you've covered all the cases mentions in that PR.
> 
> I fixed all the todos. But I didn't understand what is your point about this line:
> 
> ` // TODO: write a test util to update config.`
> 
> I think it will be cleaner to update the config in the test file.

I meant creating a `test-utils/store.ts` and the wrtie a util called `updateConfig`  to be used whenever we need to update config.
Instead of this:
```ts
   appStoreState.config.from = {
      ...appStoreState.config.from,
      pinnedTokens: [
        {
          symbol: rangoToken.symbol,
          address: rangoToken.address,
          blockchain: rangoToken.blockchain,
        },
      ],
    };
```
It will be like:
```ts
updateConfig(appStoreState, {
 from: {
    pinnedTokens: [ ....]
 }
})
```

yeager-eren commented on widget/embedded/src/store/slices/data.test.ts:
```diff
@@ -0,0 +1,345 @@
+});
+
+describe('check sorting tokens is working correctly in app store', () => {
+  test('put pinned tokens first', async () => {
```
Please remove `async` if they aren't needed.

nikaaru commented on widget/embedded/src/setup-vitest.ts:
```diff
@@ -0,0 +1,4 @@
+import { vi } from 'vitest';
+
+// to make it works like Jest (auto-mocking)
+vi.mock('zustand');
```
Done
